app/grandchallenge/algorithms/static/algorithms/js/session.js

```diff
@@ -0,0 +1,154 @@
+"use strict";
+
+const timeout = 5000;
+
+function getUploadSessionStatus(statusUrl, cards) {
+    // Checks on the status of the Session (queued, running, started, etc)
+    fetch(statusUrl)
+        .then(response => response.json())
+        .then(session => handleUploadSessionStatus(statusUrl, cards, session.status, session.image_set));
+}
+
+function handleUploadSessionStatus(statusUrl, cards, status, imageUrls) {
+    switch (status.toLowerCase()) {
+        case "queued":
+        case "re-queued":
+            setCardAwaitingMessage(cards.imageImport, status);
+            setTimeout(function () {
+                getUploadSessionStatus(statusUrl, cards)
+            }, Math.floor(Math.random() * timeout) + 100);
+            break;
+        case "started":
+            setCardActiveMessage(cards.imageImport, status);
+            setTimeout(function () {
+                getUploadSessionStatus(statusUrl, cards)
+            }, Math.floor(Math.random() * timeout) + 100);
+            break;
+        case "succeeded":
+            let msg = "Imported " + imageUrls.length + " Image";
```

Comment:
you could also use backticks here:
```suggestion
            let msg = `Imported ${imageUrls.length} Image`;
```

Reply:
Thanks! I was desperately searching on "how do I f-string in js" yesterday...